src/poetry/utils/extras.py

```diff
@@ -9,12 +9,13 @@
     from collections.abc import Sequence
     from typing import Mapping
 
+    from packaging.utils import NormalizedName
     from poetry.core.packages.package import Package
 
 
 def get_extra_package_names(
     packages: Sequence[Package],
-    extras: Mapping[str, list[str]],
+    extras: Mapping[str, list[NormalizedName]],
```

Comment:
If we believe this change, then the call to `canonicalize_name` a few lines later is redundant and should be removed

I'm not completely sure that I do believe it: some callers populate this parameter by reading from the locker: is `lock_data.get("extras", {})` definitely returning `NormalizedName`s?

I think the answer is that it is if the lockfile was written by a sufficiently recent poetry, possibly it isn't otherwise.

Reply:
(not to mention that the text in the locker is often like
```toml
[package.extras]
foo = ["foo[bar] (>=1.5)"]
```
surely that doesn't work in this code at all (regardless of typing)?!?

Reply:
Actually, poetry 1.2 should be able to read lock files generated by poetry 1.1. Thus, I think we should keep it.

Reply:
both poetry 1.1 and 1.2 write un-normalized strings to the lockfile and so the annotation is flat wrong.

Reply:
Ahh, now I see.